### PR TITLE
Formalize shared selectors as skolem functions

### PR DESCRIPTION
**Pull request by ajreynol**
This is work towards properly printing shared selectors in external proofs.

### CODE REVIEW

File: src/expr/skolem_manager.h [hunk 1]
```diff
@@ -231,7 +233,8 @@ class SkolemManager
    */
   Node mkSkolemFunction(SkolemFunId id,
                         TypeNode tn,
-                        Node cacheVal = Node::null());
+                        Node cacheVal = Node::null(),
+                        int flags = NodeManager::SKOLEM_DEFAULT);
```
Comment (HanielB) [inline src/expr/skolem_manager.h:237]: use unsigned?

Comment (ajreynol) [reply src/expr/skolem_manager.h:237]: This is to match `NodeManager::` flags.  We shouldn't mix types here until those flags are fully refactored at once. (Note that `int flags` appears as arguments to other methods in this class for the same reason.

Comment (HanielB) [reply src/expr/skolem_manager.h:237]: I see, sounds good

**Review by HanielB — APPROVED** [review]
LGTM

**Pull request merged**